controllers/cards.js

```diff
@@ -19,18 +24,24 @@ module.exports.createCard = (req, res) => {
     .catch(() => res.status(500).send({ message: 'Не удается создать карточку' }));
 };
 
-module.exports.deleteCard = (req, res) => {
+module.exports.deleteCard = (req, res, next) => {
   const { cardId } = req.params;
-  const ownerId = req.user._id;
-  Card.findById(cardId)
+  if (!ObjectId.isValid(cardId)) {
+    return res.status(404).send({ message: 'not found' });
+  }
+  Card.findById(req.params.cardId)
     .then((card) => {
-      if (card.owner.toString() === ownerId) {
-        Card.findByIdAndRemove(cardId)
-          .then((card) => res.send({ data: card }))
-          .catch(() => errorSend(res));
+      if (card) {
+        if (card.owner.toString() === req.user._id) {
```

Comment:
Можно лучше: использовать метод `.equals()` при сравнении.
Экземпляр `card.owner` является экземпляром класса `Document`, у него есть метод [`.equals()`](https://mongoosejs.com/docs/api.html#document_Document-equals), который позволяет сравнивать данные. Это нужно затем, чтобы разработчики приложений не смотрели на типы.

Те так:
```suggestion
        if (card.owner.equals(req.user._id)) {
```